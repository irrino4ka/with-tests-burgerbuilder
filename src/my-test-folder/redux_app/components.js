import React from 'react';

// export function Todo(props) {
//   const { todo } = props;

//   if (todo.isDone) {
//     return <strike>{todo.text}</strike>;
//   } else {
//     return <span>{todo.text}</span>;
//   }

// }
// export function TodoBold(props) {
//   const { todo } = props;

//   if (todo.isBold) {
//     return <b>{todo.text}</b>;
//   } else {
//     return <span>{todo.text}</span>;
//   }
// }

export function TodoCase(props) {
  console.log('case', props);
  
  const { todo } = props;
  if (todo.isBold) {
    return <b>{todo.text}</b>;
  } else if (todo.isDone) {
    return <strike>{todo.text}</strike>;
  } else {
    return <span>{todo.text}</span>;
  }
}

export function TodoList(props) {
  const { todos, toggleTodo, addTodo, boldTodo } = props;
  console.log('todos');

  const onSubmit = (event) => {
    event.preventDefault()
    console.log('submit', event);
    console.log(event.target.querySelector('input').value);
    const input = event.target;
    const text = event.target.querySelector('input').value;
    const isLongEnough = text.length > 0;
    if (isLongEnough) {
      event.target.querySelector('input').value = '';
      addTodo(text);
    } else {
      alert('Input correct value')
    }
  };

  const toggleClick = id => event => toggleTodo(id);
  const boldClick = id => event => boldTodo(id);

  return (
    <div className='todo'>
      <form onSubmit={onSubmit} >
        <input type='text'
          className='todo__entry'
          placeholder='Add todo'
        />
        <input className='todo__button' type="submit" value="Add" />
      </form>
      <ul className='todo__list'>
        {todos.map(t => (
          <li key={t.get('id')}
            className='todo__item'
          >
            {/* <p className='TOdo_class' onClick={toggleClick(t.get('id'))} ><Todo todo={t.toJS()} /></p>
            <p className='bold_class' onClick={boldClick(t.get('id'))}><TodoBold todo={t.toJS()} /></p> */}
            <TodoCase todo={t.toJS()} />
            <button  className='todo__button'  onClick={toggleClick(t.get('id'))}> Done</button>
            <button  className='todo__button'  onClick={boldClick(t.get('id'))}>Bold</button>
          </li>
        ))}
      </ul>
    </div>
  );
}

