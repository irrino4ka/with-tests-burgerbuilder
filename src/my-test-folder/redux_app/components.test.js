import { TodoCase, TodoList} from './components';
import React from 'react';
import { shallow, mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('test TodoCase', () => {

    let wrapper;
    beforeEach(() => {
        wrapper = mount(<TodoCase todo="isBold" />);
    });

    it('should test component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should have property set', () => {
        expect(wrapper.props().todo).not.toBe('isDone');
    });

    it('should have 1 property', () => {
        expect(wrapper.props()).toBeDefined();
    })
})