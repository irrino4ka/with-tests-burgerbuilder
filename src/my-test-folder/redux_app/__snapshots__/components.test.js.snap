// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test TodoCase should test component 1`] = `
ReactWrapper {
  "length": 1,
  Symbol(enzyme.__unrendered__): <TodoCase
    todo="isBold"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "todo": "isBold",
    },
    "ref": null,
    "rendered": Object {
      "instance": <span />,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
      },
      "ref": null,
      "rendered": Array [
        undefined,
      ],
      "type": "span",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "todo": "isBold",
      },
      "ref": null,
      "rendered": Object {
        "instance": <span />,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": Array [
          undefined,
        ],
        "type": "span",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
